sonar-db/src/main/java/org/sonar/db/compute/AnalysisReportDao.java

```diff
@@ -69,6 +69,10 @@ public AnalysisReportDto pop(DbSession session) {
     return tryToPop(session, reportId);
   }
 
+  public long countPending(DbSession session) {
+    return mapper(session).selectAvailables(PENDING, WORKING).size();
```

Comment:
This was my though when I first saw a call to this `selectAvailable` method. Method actually uses the second parameter to check there is no other report for a given project with a report in PENDING state which has status WORKING.